src/cloudflare/internal/streaming-base64.ts

```diff
@@ -0,0 +1,194 @@
+import base64 from 'cloudflare-internal:base64';
+
+function base64Error(cause: unknown): Error {
+  // @ts-expect-error: Error.isError seems to be missing from types?
+  const isError: (_: unknown) => boolean = Error.isError; // eslint-disable-line @typescript-eslint/no-unsafe-assignment
+  if (isError(cause)) {
+    const e = cause as Error;
+    return new Error(`base64 error: ${e.message}`, { cause });
+  } else {
+    return new Error('unknown base64 error');
+  }
+}
+
+function toBase64(input: Uint8Array): Uint8Array {
+  return new Uint8Array(base64.encodeArray(input));
```

Comment:
You can use the following since encodeArray returns an ArrayBuffer, which will be faster afaik:

```
new Uint8Array(buffer, byteOffset, length)
```

Reply:
I think these are the same things - `new Uint8Array(arrayBuffer)` will point at the same memory:
```
> ab = new ArrayBuffer(1)
ArrayBuffer { [Uint8Contents]: <00>, byteLength: 1 }
> u8a = new Uint8Array(ab)
Uint8Array(1) [ 0 ]
> u8a.set([1], 0)
undefined
> u8a
Uint8Array(1) [ 1 ]
> ab
ArrayBuffer { [Uint8Contents]: <01>, byteLength: 1 }
```